frontend/src/components/ButtonSelect.js

```diff
@@ -100,7 +100,7 @@ function ButtonSelect(props) {
       updateDateRange(range);
     }
 
-    onApply(selectedItem);
```

Comment:
I was looking at this component the other day and thought it might be time for a refactor. It tries to handle date selection along with non-date selection and it feels like it causes users of the component to work harder then they should. As a user of this component I feel like I should only need to pass one callback function for the value that was selected. Right now I need to pass two in, one for when the date changes and another for when the selected value changes. When I want to use this component to select dates I don't _think_ I care if a custom range is selected or if the last 30 days was selected, I only care about the actual date range.

We currently double call out to the API on the regional dashboard and tracking down the issue this component seemed to be the root cause. Refactoring this component would allow us to remove some `useEffects` that were changing state on initial load causing the API to get called twice.

Reply:
Makes sense, it does feel unwieldy and, as you say, it doesn't make sense to maintain the selected option and the date range callbacks

Reply:
Thanks for the refactor! Looks a lot easier to use the `ButtonSelect` and `DateRangeSelect` components!